src/module.c

```diff
@@ -696,7 +731,7 @@ static int internalAdd(RedisModuleCtx *ctx,
     }
 
     if (timestamp <= series->lastTimestamp && series->totalSamples != 0) {
-        if (SeriesUpsertSample(series, timestamp, value) != REDISMODULE_OK) {
+        if (SeriesUpsertSample(series, timestamp, value, dp_override) != REDISMODULE_OK) {
```

Comment:
```suggestion
    if (dp_override != DP_FIRST && dp_override != DP_BLOCK) { 
            if (SeriesUpsertSample(series, timestamp, value, dp_override) != REDISMODULE_OK) {
```

Reply:
@ashtul that's not what we want to do, we want to block *only* if its a duplicate sample not every upsert.

Reply:
That is true! 👍🏻
@K-Jo do you want to have an option to block out of order even if not a duplicate?